babel/messages/pofile.py

```diff
@@ -80,6 +80,50 @@ def denormalize(string: str) -> str:
         return unescape(string)
 
 
+def _extract_locations(line: str) -> list[str]:
+    """Extract locations from location comments.
+
+    Locations are extracted while properly handling First Strong
+    Isolate (U+2068) and Pop Directional Isolate (U+2069), used by
+    gettext to enclose filenames with spaces and tabs in their names.
+    """
+    if "\u2068" not in line and "\u2069" not in line:
```

Comment:
TBH, I think looking for only one of these is enough for us – it should be safe to assume that a .po writer that writes these characters to begin with will write both of them.

```suggestion
    if "\u2068" not in line:
```

Reply:
When checking only for First Strong Isolate and parsing a corrupt file, one might end up with the Pop Directional Isolate being part of the extracted locations:

```
>>> "first location\u2069 second_location".lstrip().split()
['first', 'location\u2069', 'second_location']
```

I believe raising an error would be more appropriate in that situation.

Reply:
Practically speaking, I don't think files will (often or ever) be corrupt like that.

But I measured having both checks and having only one check, and it's practically the same speed, all good.